### ISSUE TITLE
compiler can't find sratom

### ISSUE THREAD

**Issue by magnetophon** [OpenMusicKontrollers/synthpod#55]
When I try to build on NixOS, I get:
```
building
build flags: -j4 -l4
[53/139] Compiling C object app/libsynthpod_app.a.p/synthpod_app.c.o.c.o[K
FAILED: app/libsynthpod_app.a.p/synthpod_app.c.o
gcc -Iapp/libsynthpod_app.a.p -Iapp -I../app -I../include -I../canvas.lv2 -I../xpress.lv2 -I../osc.lv2 -I../ext_ui.lv2 -I../ardour.lv2 -I../varchunk -I../cross_clock -I/nix/store/f5lkbncd6zgdvawqm2hj1ap7z3k1wy6q-lv2-1.18.0/include -I/nix/store/87ayy7b4lk15w7xmrifjcqz4j839x5nm-lilv-0.24.10/include/lilv-0 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu11 -DUSE_DYNAMIC_PARALLELIZER -DPUGL_HAVE_GL -DUSE_CAIRO_CANVAS -DLILV_0_22 -D_GNU_SOURCE '-DSYNTHPOD_VERSION="0.1.6335"' -DSYNTHPOD_MINOR_VERSION=1 -DSYNTHPOD_MICRO_VERSION=6335 '-DSYNTHPOD_BUNDLE_DIR="/nix/store/1w94c79b2yz8m1dkd50s9kj7jjxccz1b-synthpod-unstable-2020-12-04/lib/synthpod/lv2/"' '-DSYNTHPOD_PLUGIN_DIR="/nix/store/1w94c79b2yz8m1dkd50s9kj7jjxccz1b-synthpod-unstable-2020-12-04/lib/lv2/synthpod.lv2/"' -DHAS_BUILTIN_ASSUME_ALIGNED -fPIC -pthread -fvisibility=hidden -Wno-attributes -Wno-unused-function -Wno-unused-variable -ftree-vectorize -ftree-vectorizer-verbose=1 -MD -MQ app/libsynthpod_app.a.p/synthpod_app.c.o -MF app/libsynthpod_app.a.p/synthpod_app.c.o.d -o app/libsynthpod_app.a.p/synthpod_app.c.o -c ../app/synthpod_app.c
In file included from ../app/synthpod_app.c:18:
../app/synthpod_app_private.h:33:10: fatal error: sratom/sratom.h: No such file or directory
   33 | #include <sratom/sratom.h>
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
[56/139] Compiling C object sandbox_ui.lv2/libsbox_slave.a.p/sandbox_slave.c.o[K
../sandbox_ui.lv2/sandbox_slave.c: In function '_file_read':
../sandbox_ui.lv2/sandbox_slave.c:323:2: warning: ignoring return value of 'read', declared with attribute warn_unused_result [-Wunused-result]
  323 |  read(fd, buf, *len);
      |  ^~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
builder for '/nix/store/zysr8qqqvq5x3d1cs4niahq9flvprkzx-synthpod-unstable-2020-12-04.drv' failed with exit code 1
error: build of '/nix/store/zysr8qqqvq5x3d1cs4niahq9flvprkzx-synthpod-unstable-2020-12-04.drv' failed
```

Of course, sratom is in the PKG_CONFIG_PATH.
Since NixOS is a bit odd when it comes to dependencies, I tried hardpatching the location of sratom.
When I do that, it can't find serd inside sratom:
```
building
build flags: -j4 -l4
[54/139] Compiling C object app/libsynthpod_app.a.p/synthpod_app.c.oatrix.c.o
FAILED: app/libsynthpod_app.a.p/synthpod_app.c.o
gcc -Iapp/libsynthpod_app.a.p -Iapp -I../app -I../include -I../canvas.lv2 -I../xpress.lv2 -I../osc.lv2 -I../ext_ui.lv2 -I../ardour.lv2 -I../varchunk -I../cross_clock -I/nix/store/f5lkbncd6zgdvawqm2hj1ap7z3k1wy6q-lv2-1.18.0/include -I/nix/store/87ayy7b4lk15w7xmrifjcqz4j839x5nm-lilv-0.24.10/include/lilv-0 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu11 -DUSE_DYNAMIC_PARALLELIZER -DPUGL_HAVE_GL -DUSE_CAIRO_CANVAS -DLILV_0_22 -D_GNU_SOURCE '-DSYNTHPOD_VERSION="0.1.6335"' -DSYNTHPOD_MINOR_VERSION=1 -DSYNTHPOD_MICRO_VERSION=6335 '-DSYNTHPOD_BUNDLE_DIR="/nix/store/x8p5n19ymsbnlyaxwvbfby5cihci9n7a-synthpod-unstable-2020-12-04/lib/synthpod/lv2/"' '-DSYNTHPOD_PLUGIN_DIR="/nix/store/x8p5n19ymsbnlyaxwvbfby5cihci9n7a-synthpod-unstable-2020-12-04/lib/lv2/synthpod.lv2/"' -DHAS_BUILTIN_ASSUME_ALIGNED -fPIC -pthread -fvisibility=hidden -Wno-attributes -Wno-unused-function -Wno-unused-variable -ftree-vectorize -ftree-vectorizer-verbose=1 -MD -MQ app/libsynthpod_app.a.p/synthpod_app.c.o -MF app/libsynthpod_app.a.p/synthpod_app.c.o.d -o app/libsynthpod_app.a.p/synthpod_app.c.o -c ../app/synthpod_app.c
In file included from ../app/synthpod_app_private.h:33,
                 from ../app/synthpod_app.c:18:
/nix/store/zr9v1sm6z77q83hgdw4jbi28vhyqsfhp-sratom-0.6.4/include/sratom-0/sratom/sratom.h:27:10: fatal error: serd/serd.h: No such file or directory
   27 | #include "serd/serd.h"
      |          ^~~~~~~~~~~~~
compilation terminated.
[57/139] Compiling C object sandbox_ui.lv2/libsbox_slave.a.p/sandbox_slave.c.o[K
../sandbox_ui.lv2/sandbox_slave.c: In function '_file_read':
../sandbox_ui.lv2/sandbox_slave.c:323:2: warning: ignoring return value of 'read', declared with attribute warn_unused_result [-Wunused-result]
  323 |  read(fd, buf, *len);
      |  ^~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
builder for '/nix/store/4cv5hljgm6ffc7gn7ja492m1bmlpkqxf-synthpod-unstable-2020-12-04.drv' failed with exit code 1
error: build of '/nix/store/4cv5hljgm6ffc7gn7ja492m1bmlpkqxf-synthpod-unstable-2020-12-04.drv' failed
```

This might very well be a NixOS issue.

**Comment by ventosus** [comment]
thanks for reporting.

sratom is not looked for explicitely in the build script, it is a hard dependency of liblilv, maybe the latter is somehow broken in NixOS?

Here on Parabola I get:

    $ pkg-config --cflags lilv-0
    -I/usr/include/lilv-0 -I/usr/include/serd-0 -I/usr/include/sord-0 -I/usr/include/sratom-0

**Comment by magnetophon** [comment]
> sratom is not looked for explicitely in the build script

Maybe not there, but it is here, iiuc: https://github.com/OpenMusicKontrollers/synthpod/blob/master/app/synthpod_app_private.h#L33

I'm not sure if I understand you. Does it matter if the buildscript checks or not, when the .h is needed anyway?

> liblilv, maybe the latter is somehow broken in NixOS?

I don't know if lilv is broken in NixOS, but I do know there are plenty of packages using it successfully in NixOS.

When I run ``pkg-config --cflags lilv-0`` in the build environment , I get:
```

-I/nix/store/87dsdjhffjxpmmacx3r59xb5dqpq8wc9-lv2-1.18.0/include -I/nix/store/snxbdgdfvqd48lbg800lq6vw1j8432mm-lilv-0.24.10/include/lilv-0
```
Is that bad?

**Comment by ventosus** [comment]
maybe fixed in 333829737728b7883aae72928c4b7d3505075a6e

**Comment by magnetophon** [comment]
Fixed indeed.
Thank you!
